Automattic/sensei
# Extract ability to assign a product to a Course

### Changes Proposed in this PR
This moves logic from the WP admin's Edit Lesson (when creating a course) and Edit Course screens out of this plugin. 

### New Hooks
- Filters:
  - `includes/class-sensei-lesson.php`: `sensei_lesson_course_create_form_top`: Fires above the create course form used in lesson editor meta box. Filters the current generated HTML for the meta box.
  - `includes/class-sensei-lesson.php`: `sensei_lesson_course_create_form_bottom`: Fires below the create course form used in lesson editor meta box. Filters the current generated HTML for the meta box.
- Actions: 
  - `includes/class-sensei-lesson.php`: `sensei_lesson_course_created`: Fires when a new course was created from the lesson editor meta box.

### Testing Instructions
1) Deactivate WCPC.
1) From WP admin, edit or add a new course.
1) Verify Product meta box is no longer on right side.
1) Save product.
1) Verify nothing strange appears in JS console or PHP error log. Also, verify things save correctly.
1) From the list of Courses in WP Admin, verify no Product column appears. 
1) Edit or add a new lesson.
1) On the course meta box, create a new course.
1) Verify the Product dropdown doesn't appear as a field for creating the new course.
1) Create the course. 
1) Verify nothing strange appears in JS console or PHP error log. Also, verify things save correctly.
